go/arrow/flight/server.go

```diff
@@ -63,6 +104,8 @@ type server struct {
 // Alternatively, a grpc server can be created normally without this helper as the
 // grpc server generated code is still being exported. This only exists to allow
 // the utility of the helpers
+//
+// Deprecated: prefer to use NewServerWithMiddleware
```

Comment:
should this be deprecated, if implementators don't care about middleware, is there a strong reason to use that method over this one?

Reply:
There's a couple reasons:

The `NewServer` function can actually cause problems if using an auth handler + passing in grpc options that set interceptors because of how it was handled, whereas using this given the definition of the middlewares makes it explicit that interceptors should be set using the middleware rather than using the generic grpc options.

In addition, by marking this deprecated, we can eventually remove this old function and make `NewServerWithMiddleware` the actual `NewServer` function so everything is very explicit.